### PR TITLE
Continuous Integration: python-version: ["3.9", "3.10", "3.11", "3.12", "3.13"]

### PR DESCRIPTION
**Pull request by cclauss**
Test on current versions of Python.
* https://devguide.python.org/versions

- - -
This change is [Reviewable](https://reviewable.io/reviews/google/openhtf/1185)

### CODE REVIEW

**Comment by glados-verma** [comment]
A little bit of work is need to support 3.12 and newer -  I made partial [progress here](https://github.com/google/openhtf/compare/master...glados-verma:openhtf:py312). I think the only thing still needing work is making the built-in test profiling work correctly - for some reason the generated profile files seem to be empty.

**Comment by glados-verma** [comment]
> ```python
>       profile_tempfile = tempfile.NamedTemporaryFile(delete=False)
> ```
> 
> The new optional parameter in Python 3.12 is called `delete_on_close` which is different than `delete`.
> 
> * https://docs.python.org/3/whatsnew/3.12.html#tempfile

Yes, but: `If delete is false, the value of delete_on_close is ignored` from [`NamedTemporaryFile`](https://docs.python.org/3/library/tempfile.html#tempfile.NamedTemporaryFile)

File: .github/workflows/continuous_integration.yml [hunk 1]
```diff
@@ -26,7 +26,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        python-version: ["3.9", "3.10", "3.11"]
+        python-version: ["3.9", "3.11", "3.13"]
```
Comment (glados-verma) [inline .github/workflows/continuous_integration.yml:29]: Please add 3.10 and 3.12 here as well.

**Comment by glados-verma** [comment]
I'm looking into the profiling breakage under Python3.12+. The error message might be misleading.

**Comment by cclauss** [comment]
https://docs.python.org/3/whatsnew/3.12.html#distutils

**Comment by glados-verma** [comment]
The distutils related issue is already fixed in #1183 . The problem right now is that the profiling unit test fails. I have found [the cause for the breakage](https://github.com/python/cpython/issues/110770). It looks like the profiling implementation in OpenHTF isn't quite right, so I'm looking into that now.

File: .github/workflows/continuous_integration.yml [hunk 2]
```diff
@@ -45,7 +45,7 @@ jobs:
         python -m build
     - name: Test with tox
       run: |
-        python -m pip install tox tox-gh-actions
+        python -m pip install tox tox-gh-actions setuptools
```
Comment (glados-verma) [inline .github/workflows/continuous_integration.yml:48]: setuptools installation is no longer needed after #1183 , you can revert this line.

File: tox.ini [hunk 3]
```diff
@@ -14,6 +16,7 @@ deps =
     numpy
     pytest>=2.9.2
     pytest-cov>=2.2.1
+    setuptools
```
Comment (glados-verma) [inline tox.ini:19]: Also not needed

**Review by glados-verma — CHANGES_REQUESTED** [review]

File: .github/workflows/continuous_integration.yml (hunk 2 above)
Comment (cclauss) [reply .github/workflows/continuous_integration.yml:48]:
```suggestion
        python -m pip install tox tox-gh-actions
```

File: tox.ini (hunk 3 above)
Comment (cclauss) [reply tox.ini:19]:
```suggestion
```

**Review by glados-verma — APPROVED** [review]

**Comment by glados-verma** [comment]
Thanks for the PR!

**Pull request merged**